### PR TITLE
Buildcache create: change overwrite exception back to a warning as in v0.14

### LINKED ISSUE #17784: "spack buildcache create ..." fails if the a file already exists
In version v0.14, `spack buildcache create ...` would only report a warning if the build cache of a dependency already existed.

Now, on version v0.15, it returns an error and suggests I use the option `-f` to overwrite the existing files. I don't want to do that - it's slow and wastes time.

How can I get the old behaviour back?

### Steps to reproduce the issue

```console
$ git checkout releases/v0.15
$ spack buildcache create -a -u -d src_mirror speck
$ spack buildcache create -a -u -d src_mirror speck
==> Buildcache files will be output to file:///home/vagrant/src_mirror/build_cache
==> Error: 
file:///home/vagrant/src_mirror/build_cache/linux-rhel6-x86_64/gcc-7.3.1/libuuid-1.0.3/linux-rhel6-x86_64-gcc-7.3.1-libuuid-1.0.3-bqv2qs3zxsth6jn7fnrqqrmf3czxxzke.spack
exists
Use -f option to overwrite.
...
```

In v0.14, it reports lots of warnings, but is otherwise fine:
```console
$ git checkout releases/v0.14
$ spack buildcache create -a -u -d src_mirror speck
$ spack buildcache create -a -u -d src_mirror speck
...
==> creating binary cache file for package freetype@2.10.1%gcc@7.3.1 arch=linux-rhel6-x86_64 
==> Warning: 
file:///home/vagrant/src_mirror/build_cache/linux-rhel6-x86_64/gcc-7.3.1/freetype-2.10.1/linux-rhel6-x86_64-gcc-7.3.1-freetype-2.10.1-uvy2rw6w6qtekqo4ibr5zuua3zuu5ffx.spack
exists
Use -f option to overwrite.
==> creating binary cache file for package mesa-glu@9.0.1%gcc@7.3.1 arch=linux-rhel6-x86_64 
==> Warning: 
file:///home/vagrant/src_mirror/build_cache/linux-rhel6-x86_64/gcc-7.3.1/mesa-glu-9.0.1/linux-rhel6-x86_64-gcc-7.3.1-mesa-glu-9.0.1-oz4nxkut7kzrvusucccv6vh7tbsmgjwf.spack
exists
Use -f option to overwrite.
```

### Information on your system

* **Spack:** 0.15.3
* **Python:** 2.6.6
* **Platform:** linux-rhel6-sandybridge

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [ ] I have run the failing commands in debug mode and reported the output

### PR DESCRIPTION
**Pull request by gartung**
Addresses https://github.com/spack/spack/issues/17784

### CODE REVIEW

File: lib/spack/spack/cmd/buildcache.py [hunk 1]
```diff
@@ -399,9 +399,13 @@ def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
...
-        bindist.build_tarball(spec, outdir, force, make_relative,
-                              unsigned, allow_root, signing_key,
-                              rebuild_index)
+        try:
+            bindist.build_tarball(spec, outdir, force, make_relative,
+                                  unsigned, allow_root, signing_key,
+                                  rebuild_index)
+        except Exception as e:
+            tty.warn('%s' % e)
+            pass
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/buildcache.py]:
```suggestion
            tty.warn(e)
```
This should be equivalent.

File: lib/spack/spack/cmd/buildcache.py [hunk 2]
```diff
@@ -399,9 +399,12 @@ def _createtarball(env, spec_yaml=None, packages=None, add_spec=True,
...
     for spec in specs:
         tty.debug('creating binary cache file for package %s ' % spec.format())
-        bindist.build_tarball(spec, outdir, force, make_relative,
-                              unsigned, allow_root, signing_key,
-                              rebuild_index)
+        try:
+            bindist.build_tarball(spec, outdir, force, make_relative,
+                                  unsigned, allow_root, signing_key,
+                                  rebuild_index)
+        except Exception as e:
```
Comment (tgamblin) [inline lib/spack/spack/cmd/buildcache.py]: can this be made more specific?

**Review by tgamblin — CHANGES_REQUESTED** [review]

File: lib/spack/spack/cmd/buildcache.py (hunk 2 above)
Comment (gartung) [reply lib/spack/spack/cmd/buildcache.py]: Changed to NoOverwriteException

**Review by tgamblin — APPROVED** [review]

**Comment by tgamblin** [comment]
Still failing on `flake8` because `NoOverwriteException` isn't imported

**Pull request merged**